sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -632,6 +634,29 @@ object ScalaJSPluginInternal {
     }
   }
 
+  protected def javaSysProps2JSLib(javaOptions: Seq[String]): VirtualJSFile = {
+    val javaSysPropsPattern = "-D([^=]*)=(.*)".r
+    val javaSysProps = javaOptions.map {
+      case javaSysPropsPattern(propName, propValue) => (propName, propValue)
+    }
+    val defaultSysProps = List(
+      "java.vm.version" -> ScalaJSVersions.current,
```

Comment:
As @gzm0 said, this should go through linking info, as it's dependent on the tools; not the sbt plugin, and it should be available even we run in a different way.